js/phemex.js

```diff
@@ -0,0 +1,520 @@
+                    ],
+                    'post': [
+                        'orders',
+                    ],
+                    'put': [],
+                    'delete': [
+                        'orders/cancel',
+                    ],
+                },
+            },
+            'exceptions': {
+                'exact': {
+                    'Invalid API Key.': AuthenticationError,
+                    'This key is disabled.': PermissionDenied,
+                    'Access Denied': PermissionDenied,
+                    'Duplicate clOrdID': InvalidOrder,
+                    'orderQty is invalid': InvalidOrder,
+                    'Invalid price': InvalidOrder,
+                    'Invalid stopPx for ordType': InvalidOrder,
+                },
+                'broad': {
+                    'Signature not valid': AuthenticationError,
+                    'overloaded': ExchangeNotAvailable,
+                    'Account has insufficient Available Balance': InsufficientFunds,
+                    'Service unavailable': ExchangeNotAvailable, // {"error":{"message":"Service unavailable","name":"HTTPError"}}
+                },
+            },
+            'precisionMode': DECIMAL_PLACES,
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        const precisions = {
+            'BTCUSD': {
+                'amount': 0,
+                'price': 1,
+                'value': 8,
+            },
+            'ETHUSD': {
+                'amount': 0,
+                'price': 2,
+                'value': 2,
+            },
+            'XRPUSD': {
+                'amount': 0,
+                'price': 4,
+                'value': 2,
+            },
+        };
+        const method = 'publicGetExchangePublicProducts';
+        const response = await this[method] (params);
+        const products = this.safeValue (response, 'data');
+        const productsCount = products.length;
+        const result = [];
+        for (let i = 0; i < productsCount; i++) {
+            const market = this.parseMarket (products[i], precisions);
+            result.push (market);
+        }
+        return result;
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        const method = 'privateGetAccountsAccountPositions';
+        const btcResponse = await this[method] ({ 'currency': 'BTC' });
+        const usdResponse = await this[method] ({ 'currency': 'USD' });
+        const btcAccount = this.safeValue (this.parseResponse (btcResponse), 'account');
+        const btcBalanceEv = this.safeValue (btcAccount, 'accountBalanceEv');
+        const btcTotalUsedBalanceEv = this.safeValue (btcAccount, 'totalUsedBalanceEv');
+        const usdAccount = this.safeValue (this.parseResponse (usdResponse), 'account');
+        const usdBalanceEv = this.safeValue (usdAccount, 'accountBalanceEv');
+        const usdTotalUsedBalanceEv = this.safeValue (usdAccount, 'totalUsedBalanceEv');
+        return {
```

Comment:
Can you update the returned balance structure?

Missing `total`, `'free', and `used` on the top-level object returned.

See here: https://github.com/ccxt/ccxt/wiki/Manual#balance-structure

Reply:
Also: https://github.com/ccxt/ccxt/wiki/Requirements )

Reply:
already added total, free, and used to balance structure, please review again.